30-helm/s3-sidecar/pd-archive-backup-to-s3.yaml

```diff
@@ -0,0 +1,206 @@
+############################################################
+# Ping Identity
+# This file is an extension of 30-helm/pingdirectory-backup/pingdirectory-periodic-backup.yaml
+# The primary changes are the addition of an archive script & the assumption that permissions exist
+# for use by the archive script either through environment variables or an IAM role.
+
+############################################################
+# Archiving a backup file example to S3
+############################################################
+# Note that the storage of the backup files in this example
+# is shown FOR DEMONSTRATION PURPOSES ONLY.
+# Your particular implementation will depend on the desired storage
+# and the cloud environment (S3, etc.).
+configMaps:
+  pingdirectory-archive:
+    data:
+      archive.sh: |-
+        #!/bin/sh
+        set -e
+
+        # This script assumes it is only being run on a single pod of the topology
+
+        # Run the archive command
+        # Assumes credentials are set as environment variables or that the Pod 
+        # has an associated IAM role that allows it to write to the S3 bucket
+        aws s3 cp /opt/pingdirectory-backup s3://<your-s3-bucket-name>/<path>/ --recursive
+  pingdirectory-backup:
+    data:
+      backup.sh: |-
+        #!/bin/sh
+        set -e
+
+        # This script assumes it is only being run on a single pod of the topology
+
+        # Create a directory for the backup
+        mkdir -p /opt/pingdirectory-backup
+        # Run the backup and archive the files
+        # DEMONSTRATION ONLY! NO ERROR CHECKING OR CONTROLS ARE IN PLACE
+        backup --backupDirectory /opt/pingdirectory-backup --backendID userRoot --retainPreviousFullBackupAge '1 day' && /opt/in/archive.sh
+  pingdirectory-restore:
+    data:
+      restore.sh: |-
+        #!/bin/sh
+
+        # Read admin password from script argument
+        if [ "$#" -ne 1 ]; then
+          echo "Expected 1 parameter: admin user password."
+          exit 1
+        fi
+        _adminPassword="${1}"
+
+        # TODO implement based on the environment: pull down backup files (from S3 or other storage) to SERVER_RESTORE_DIR
```

Comment:
If we're going to show how the archive step works we probably should include the logic to pull from s3 here

Reply:
added in latest commit